mautrix_signal/__main__.py

```diff
@@ -196,5 +202,39 @@ async def manhole_global_namespace(self, user_id: UserID) -> Dict[str, Any]:
             "Puppet": Puppet,
         }
 
+    async def _notify_bridge_blocked(self, is_blocked: bool = True) -> None:
+        msg = self.config["bridge.limits.block_ends_notification"]
+        if is_blocked:
+            msg = self.config["bridge.limits.block_begins_notification"]
+            next_notification = self.last_blocking_notification + self.config["bridge.limits.block_notification_interval_seconds"]
+            # We're only checking if the block is active, since the block-end notification will not be resent and we want it ASAP
+            if next_notification > int(time.time()):
+                return
+            self.last_blocking_notification = int(time.time())
+
+        admins = list(map(lambda entry: entry[0],
+            filter(lambda entry: entry[1] == 'admin',
+                self.config["bridge.permissions"].items()
+            )
+        ))
+        if len(admins) == 0:
+            self.log.debug('No bridge admins to notify about the bridge being blocked')
+            return
+
+        self.log.debug(f'Notifying bridge admins ({",".join(admins)}) about bridge being blocked')
```

Comment:
Might be just a foot gun (aka no vulnerability that's usable from the outside), but I do wonder what happens if the `admins` array contains a string with weird Unicode characters – assuming `log.debug` doesn't escape / wrap everything in JSON.

Reply:
Those are just mxid's, so I think we'd have bigger problems if they contained weird unicode ;) Not sure if that needs to be within our "threat model" here.